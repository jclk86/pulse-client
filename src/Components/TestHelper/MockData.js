export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com",
  image_url:
    "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
  profile: "test profile"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_category: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    id: 1,
    username: mockArticle.author.username
  }
};

export const mockUsers = [
  {
    username: "Pop123",
    password: "Password123!",
    fullname: "Bob Smith",
    email: "BSmith@gmail.com",
    image_url:
      "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    profile: "test profile"
  },
  {
    username: "testuser2",
    password: "Password123!",
    fullname: "Test User",
    email: "TUser@gmail.com",
    image_url:
      "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
    profile: "test profile"
  }
];

export const mockArticles = [
  {
    title: "test article one",
    content: "test content",
    article_category: "News",
    id: 1,
    date_created: "2019-10-14T21:49:20.570Z",
    author: {
      username: mockUsers[0].username
    }
  },
  {
    title: "test article two",
    content: "test content",
    article_category: "News",
    id: 2,
    date_created: "2019-10-14T21:49:20.570Z",
    author: {
      username: mockUsers[1].username
    }
  }
];

export const categories = [
  { name: "Diary" },
  { name: "Guide" },
  { name: "Advice" },
  { name: "Interview" },
  { name: "Random" },
  { name: "News" }
];
