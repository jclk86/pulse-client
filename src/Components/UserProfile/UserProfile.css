.container_home_btn {
  width: 100%;
  margin: 0 auto;
}

.user_profile_home_btn {
  display: block;
  border: none;
  background-color: white;
  text-decoration: underline;
  padding-left: 0;
  margin-bottom: 10px;
}

.user_profile_home_btn:hover {
  cursor: pointer;
}

.user_profile_home_btn:focus {
  outline-width: 0;
}

.container_user_profile_info {
  display: flex;
  flex-direction: column;
}

.container_user_profile_image {
  width: 100%;
  margin: 0 auto;
}

.profile_image {
  width: 100%;
  height: 300px;
  object-fit: cover;
  margin-bottom: 10px;
  object-position: 50% 50%;
}

.user_profile_header {
  margin-top: 0;
}

@media only screen and (min-width: 568px) {
  .container_user_profile {
    max-width: 800px;
    margin: 0 auto;
  }

  .container_user_profile_section {
    display: flex;
    flex-wrap: wrap;
  }

  .container_home_btn {
    max-width: 1400px;
    margin-bottom: 50px;
  }

  .container_user_profile_image {
    margin-left: 0;
    width: 40%;
  }

  .container_user_profile_info {
    width: 50%;
  }
}

@media only screen and (max-width: 500px) {
  .container_user_profile_image {
    width: 100%;
    margin: 0 auto;
  }
}
