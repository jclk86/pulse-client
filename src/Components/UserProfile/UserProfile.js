import React, { Component } from "react";
import { NavLink, withRouter } from "react-router-dom";
import { Button } from "../Utils/Utils";
import UserApiService from "../../Services/user-api-service";
import UserContext from "../../Context/UserContext";
import "./UserProfile.css";

class UserProfile extends Component {
  static contextType = UserContext;
  componentDidMount() {
    const { username } = this.props.match.params;
    UserApiService.getUserProfile(username).then(this.context.setUser);
  }

  render() {
    const { user } = this.context;

    return (
      <div className="container_user_profile">
        <Button
          role="navigation"
          className="user_profile_return_btn"
          onClick={this.props.history.goBack}
        >
          Home
        </Button>
        <img
          src={user.image_url}
          alt={user.fullname}
          className="profile_image"
        ></img>
        <h1>{user.fullname}</h1>
        <p>{user.profile}</p>
        <p>Last logged in from: {user.location}</p>
      </div>
    );
  }
}

export default withRouter(UserProfile);
