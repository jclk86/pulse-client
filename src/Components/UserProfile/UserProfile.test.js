import React from "react";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import UserProfile from "./UserProfile";
import AuthApiService from "../../Services/auth-api-service";
import { mockUser } from "../TestHelper/MockData";

describe("UserProfile Component", () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });
  });

  it("renders UserProfile component without issues", () => {
    const wrapper = shallow(
      <BrowserRouter>
        <UserProfile user={mockUser}></UserProfile>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
