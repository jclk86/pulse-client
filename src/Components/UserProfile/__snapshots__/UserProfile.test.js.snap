// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Component renders UserProfile component without issues 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <withRouter(UserProfile)
    user={
      Object {
        "email": "BSmith@gmail.com",
        "fullname": "Bob Smith",
        "image_url": "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
        "password": "Password123!",
        "profile": "test profile",
        "username": "Pop123",
      }
    }
  />
</Router>
`;
